### PR TITLE
mgr/orchestrator: Add host mon mgr management to interface

### PR DESCRIPTION
**Pull request by sebastian-philipp**
As the SSH orchestrator still needs some minor tweaks, I'd like to merge the host management interface.

1. Changed the interface methods to return strings in order to make the result more generic and less error prone. 
2. Add tests. 
3. Some minor documentation tweaks. 
4. removed label form `add host`

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/orchestrator.py [hunk 1]
```diff
@@ -297,6 +297,26 @@ def remove_osds(self, osd_ids):
         """
         raise NotImplementedError()
 
+    def update_mgrs(self, num, hosts):
+        # type: (int, List[str]) -> WriteCompletion
```
Comment (jmolmo) [inline src/pybind/mgr/orchestrator.py]: `S/WriteCompletion/orchestrator.WriteCompleteion/`?

File: src/pybind/mgr/orchestrator.py [hunk 2]
```diff
@@ -297,6 +297,26 @@ def remove_osds(self, osd_ids):
...
+        """
+        Update the number of cluster managers.
+
+        :param num: requested number of managers.
+        :param hosts: list of hosts (optional)
+        """
+        raise NotImplementedError()
+
+    def update_mons(self, num, hosts):
+        # type: (int, List[Tuple[str,str]]) -> WriteCompletion
```
Comment (jmolmo) [inline src/pybind/mgr/orchestrator.py]: ditto

**Comment by jmolmo** [comment]
LGTM

**Comment by jtlayton** [comment]
Rook still needs some work in the mon scaling area, but we will eventually wire up the "update" commands.

**Comment by noahdesu** [comment]
Rebased ssh orch. on top of this and everything is working.

**Comment by sebastian-philipp** [comment]
QA success:

* http://pulpito.ceph.com/swagner-2019-02-07_13:52:41-rados:mgr-wip-swagner-testing-distro-basic-mira/

**Pull request merged**